tx.go

```diff
@@ -142,12 +143,17 @@ func (tx *tx) CreateMapReduceJobs(stmt *influxql.SelectStatement, tagKeys []stri
 
 			// create mappers for each shard we need to hit
 			for _, sg := range shardGroups {
-				if len(sg.Shards) != 1 { // we'll only have more than 1 shard in a group when RF < # servers in cluster
-					// TODO: implement distributed queries.
-					panic("distributed queries not implemented yet and there are too many shards in this group")
-				}
 
-				shard := sg.Shards[0]
+				// pick a shard to query
+				shard := sg.Shards[rand.Intn(len(sg.Shards))]
```

Comment:
I am also curios how will it behave with 3 nodes and replicaN = 2

https://github.com/influxdb/influxdb/blob/master/server.go#L1112